docs/en/building/supplemental.md

```diff
@@ -0,0 +1,541 @@
+
+## Automatic bundling using routes
+
+By default Dojo will create bundles based on the routes used by your application. In order to do this several rules must be followed.
+
+1.  `src/routes.ts` must have a default export containing the routing configuration
+1.  Widgets must be the default export of their module
+1.  `Outlet`s `render` function must use inline functions
+
+> src/routes.ts
+
+```ts
+export default [
+	{
+		path: 'home',
+		outlet: 'home',
+		defaultRoute: true
+	},
+	{
+		path: 'about',
+		outlet: 'about'
+	},
+	{
+		path: 'profile',
+		outlet: 'profile'
+	}
+];
+```
+
+> src/App.ts
+
+```ts
+export default class App extends WidgetBase {
+	protected render() {
+		return v('div', { classes: [css.root] }, [
+			w(Menu, {}),
+			v('div', [
+				w(Outlet, { key: 'home', id: 'home', renderer: () => w(Home, {}) }),
+				w(Outlet, { key: 'about', id: 'about', renderer: () => w(About, {}) }),
+				w(Outlet, { key: 'profile', id: 'profile', renderer: () => w(Profile, { username: 'Dojo User' }) })
+			])
+		]);
+	}
+}
+```
+
+The output will result in a separate bundle for each of the application's top level routes. In this example, there will be a main application bundle and bundles for `src/Home`, `src/About`, and `src/Profile`.
+
+To see automatic bundling in action create a new application using [@dojo/cli-create-app](https://github.com/dojo/cli-create-app/) and run `npm run build`. Dojo will automatically create bundles along the various routes in the sample application.
+
+## Manually specifying bundles
+
+Bundles may be manually specified in the `.dojorc` configuration file. This can be useful for breaking down an application into smaller bundles when automatic route bundling isn't sufficient.
+
+The `bundles` feature is part of the build app command. The configuration is comprised of a map of bundle names followed by a list of files or globs to match.
+
+For example, this configuration will bundle `About` and `Profile` together in a bundle named `additional.[hash].js`. Widget modules defined used with `w()` will be automatically converted to a lazily imported, local registry item in the parent widget. This provides a mechanism for declarative code splitting in your application.
+
+> .dojorc
+
+```json
+{
+	"build-app": {
+		"bundles": {
+			"additional": ["src/widgets/About", "src/widgets/Profile"]
+		}
+	}
+}
+```
+
+If we wanted to create nls internationalization modules by locale we could use globs to ensure all files under each language directory are included.
+
+> .dojorc
+
+```json
+{
+	"build-app": {
+		"bundles": {
+			"fr": ["src/**/nls/fr/**"],
+			"de": ["src/**/nls/de/**"]
+		}
+	}
+}
+```
+
+In this case Dojo will create bundles named `fr.[hash].js` and `de.[hash].js`. For more information see [Working with message bundles](/reference-guides/i18n/working-with-message-bundles) from the Internationalization reference guide.
```

Comment:
We need to work out how linking across guides can function - ideally we can have links that are valid across the markdown files as they are structured within the framework repo, then translate those to appropriate paths when building site. Something to discuss with the site devs.